client/src/components/App.js

```diff
@@ -26,11 +27,19 @@ const App = (props) => {
     fetchCurrentUser()
   }, [])
 
+  let newPostComponent
+  if (currentUser) {
+    newPostComponent = NewSaladForm
+  } else {
+    newPostComponent = RegistrationForm
+  }
+
   return (
     <Router>
       <TopBar user={currentUser} />
       <Switch>
-        <Route exact path="/salads" component={SaladListComponent} />
+        <Route exact path="/salads/new" component={newPostComponent}></Route>
+        <Route exact path="/salads" component={SaladListComponent}></Route>
```

Comment:
neither line 41 or 42 need a closing </Route> tag. Just make the initial one self-closing like the others